### PR TITLE
#194 Simplify view model state tests

### LINKED ISSUE #194: Use chained calls from mockk
https://mockk.io/#chained-calls

Remove redundant variables by using chained calls.
`listResult` is totally redundant here
```
	val listResult: ListResult = mockk {
		every { items } returns listOf(storageItem)
	}
	every { imageReference.listAll() } returns mockk {
		coEvery { await() } returns listResult
	}
```
Above can be simplified to:
```
	every { imageReference.listAll() } returns mockk {
		coEvery { await().items } returns listOf(storageItem)
	}
```
Goal of this task is to go throughout all test files and simplify then wherever possible

**TODO:**
- [x] ImagesRepositoryTest - use chained calls
- [x] PartiesRepositoryTest - use chained calls
- [x] ImagesViewModelTest - should be parameterized
- [x] #352

### PR DESCRIPTION
**Pull request by michal-kucznerowicz**

### CODE REVIEW

File: app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartyDetailsViewModelTest.kt [hunk 1]
```diff
@@ -24,31 +21,24 @@ class PartyDetailsViewModelTest : ShouldSpec(
...
+			val partyId = 2
+			val savedStateHandle = SavedStateHandle(mapOf("partyId" to partyId))
+			val partyRemote: PartyRemote = mockk()
+			val party: Party = mockk()
 			mockkStatic(PartyRemote::toUI)
 			every { partyRemote.toUI() } returns party
-		}
-
-		afterSpec {
-			unmockkAll()
```
Comment (MarekMacko) [inline app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartyDetailsViewModelTest.kt:40]: `unmockkAll` get lost 😨

File: app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModelTest.kt [hunk 2]
```diff
@@ -23,28 +20,22 @@ class PartiesViewModelTest : ShouldSpec(
...
-				flowOf(partiesRemote) to Loaded(parties),
-				flow<List<PartyRemote>> { throw Exception() } to Error,
-			).forAll { (result, state) ->
-				every { repository.parties() } returns result
+			testState<Flow<List<PartyRemote>>, List<Party>>(
+				loadingAnswer = { returns(emptyFlow()) },
+				errorAnswer = { returns(flow { throw Exception() }) },
+				loadedAnswer = { returns(flowOf(partiesRemote)) },
+				loadedData = parties,
+			) { (state, answers) ->
```
Comment (MarekMacko) [inline app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModelTest.kt:33]: The idea seems to be interesting in general but I feel that it's too sophisticated at first glance 😄 
At this moment I would stay with previous approach of using `mockkAnswer` and adapt it to the remaining tests. 
It seems to be more complicated to use `testState` and in the end I don't see much simplification in comparison to the previous approach, even in terms of needed lines of code. 
Let's wait for others opinion and in the meantime I would think about more deeply 😄 

Anyway, big shout-out for such proposals!

Comment (bartek977) [reply app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModelTest.kt:33]: Cuz it's seems the pr is ready to merge but not all of us are convinced to this simplification, so maybe let's vote and either merge or close this pr? @michal-kucznerowicz  @MarekMacko @BorysChajdas @kmaslowiec @marcinWisnia

Comment (MarekMacko) [reply app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModelTest.kt:33]: Nice initiative @bartek977, thanks :)

Comment (michal-kucznerowicz) [reply app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModelTest.kt:33]: Merge 👍
Close 👎
@MarekMacko @bartek977 @BorysChajdas @kmaslowiec @marcinWisnia

Comment (michal-kucznerowicz) [reply app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModelTest.kt:33]: Closing PR based on voting result

**Comment by MarekMacko** [comment]
@michal-kucznerowicz Please also remove the branch

**Comment by michal-kucznerowicz** [comment]
@MarekMacko Done

**Pull request closed without merging**